evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -1024,73 +1032,362 @@ def test_component_graph_dataset_with_different_types():
+            "column_2": [1, 2, 3, 4, 5, 6, 5, 4, 3],
+            "column_3": [True, False, True, False, True, False, True, False, False],
+        }
+    )
+    y = pd.Series([1, 0, 1, 0, 1, 1, 0, 0, 0])
+    # woodwork would infer this as boolean by default -- convert to a numeric type
+    X = infer_feature_types(X, {"column_3": "integer"})
+
+    component_graph = ComponentGraph(graph)
+    # we don't have feature type selectors defined yet, so in order for the above graph to work we have to
+    # specify the types to select here.
+    # if the user-specified woodwork types are being respected, we should see the pass-through column_3 staying as numeric,
+    # meaning it won't cause a modeling error when it reaches the estimator
+    component_graph.instantiate(
+        {
+            "Select numeric col_2": {"columns": ["column_2"]},
+            "Select categorical col_1": {"columns": ["column_1"]},
+            "Pass through col_3": {"columns": ["column_3"]},
+        }
+    )
+    assert component_graph.input_feature_names == {}
+    component_graph.fit(X, y)
+
     input_feature_names = component_graph.input_feature_names
-    assert input_feature_names["Imputer"] == ["column_1", "column_2", "column_3"]
-    assert input_feature_names["OneHot"] == ["column_1", "column_2", "column_3"]
-    assert input_feature_names["DateTime"] == [
-        "column_3",
-        "column_1_a",
-        "column_1_b",
-        "column_1_c",
-        "column_1_d",
-        "column_2_1",
-        "column_2_2",
-        "column_2_3",
-        "column_2_4",
-        "column_2_5",
-        "column_2_6",
-    ]
-    assert input_feature_names["Scaler"] == [
-        "column_3",
-        "column_1_a",
-        "column_1_b",
-        "column_1_c",
-        "column_1_d",
-        "column_2_1",
-        "column_2_2",
-        "column_2_3",
-        "column_2_4",
-        "column_2_5",
-        "column_2_6",
-    ]
-    assert input_feature_names["Random Forest"] == [
-        "column_3",
-        "column_1_a",
-        "column_1_b",
-        "column_1_c",
-        "column_1_d",
-        "column_2_1",
-        "column_2_2",
-        "column_2_3",
-        "column_2_4",
-        "column_2_5",
-        "column_2_6",
-    ]
-    assert input_feature_names["Elastic Net"] == [
-        "column_3",
-        "column_1_a",
-        "column_1_b",
-        "column_1_c",
-        "column_1_d",
-        "column_2_1",
-        "column_2_2",
-        "column_2_3",
-        "column_2_4",
-        "column_2_5",
-        "column_2_6",
-    ]
-    assert input_feature_names["Logistic Regression"] == [
-        "Random Forest",
-        "Elastic Net",
+    assert input_feature_names["Random Forest"] == (
+        ["column_2", "column_1_a", "column_1_b", "column_1_c", "column_1_d", "column_3"]
+    )
+    assert mock_rf_fit.call_args[0][0].ww.logical_types["column_3"] == Integer
+    assert mock_rf_fit.call_args[0][0].ww.logical_types["column_2"] == Double
```

Comment:
Adding this to make sure we don't overwrite the types created by the standard scaler.